ocs_ci/helpers/vdbench_helpers.py

```diff
@@ -0,0 +1,696 @@
+
+    Returns:
+        dict: Dictionary mapping pod name to logs
+    """
+    pods = get_vdbench_pods(namespace, deployment_name)
+    logs_dict = {}
+
+    for pod in pods:
+        try:
+            cmd = f"oc logs -n {namespace} {pod} -c {container}"
+            logs = run_cmd(cmd)
+            logs_dict[pod] = logs
+        except CommandFailed as e:
+            log.warning(f"Failed to get logs from pod {pod}: {e}")
+            logs_dict[pod] = f"Error getting logs: {e}"
+
+    return logs_dict
+
+
+def parse_vdbench_output(logs):
+    """
+    Parse Vdbench output logs to extract performance metrics.
+
+    Args:
+        logs: Vdbench output logs
+
+    Returns:
+        dict or None: Parsed metrics or None if parsing fails
+    """
+    metrics = {}
+
+    try:
+        lines = logs.split("\n")
+        for line in lines:
+            line = line.strip()
+
+            # Parse summary statistics
+            if "avg_rate" in line and "resp" in line:
+                # Example line: "avg_rate=1234.5 resp=5.67"
+                parts = line.split()
+                for part in parts:
+                    if "=" in part:
+                        key, value = part.split("=", 1)
+                        try:
+                            metrics[key] = float(value)
+                        except ValueError:
+                            metrics[key] = value
+
+            # Parse other relevant metrics
+            if "Total" in line and "rate=" in line:
+                # Parse total statistics
+                if "rate=" in line:
+                    rate_part = line.split("rate=")[1].split()[0]
+                    try:
+                        metrics["total_rate"] = float(rate_part)
+                    except ValueError:
+                        pass
+
+        return metrics if metrics else None
+
+    except AttributeError as e:
+        log.error(f"Invalid input type for logs. Expected string. Error: {e}")
+        return None
+
+
+def monitor_vdbench_workload(workload, interval=30, duration=300):
+    """
+    Monitor a running Vdbench workload and collect metrics.
+
+    Args:
+        workload: VdbenchWorkload instance
+        interval: Monitoring interval in seconds
+        duration: Total monitoring duration in seconds
+
+    Returns:
+        list: List of metric snapshots
+    """
+    metrics_history = []
+    start_time = time.time()
+
+    while (time.time() - start_time) < duration:
+        try:
+            # Get current status
+            status = workload.get_workload_status()
+
+            # Get logs and parse metrics
+            logs_dict = get_vdbench_logs(workload.namespace, workload.deployment_name)
+
+            snapshot = {"timestamp": time.time(), "status": status, "metrics": {}}
+
+            # Parse metrics from each pod
+            for pod_name, logs in logs_dict.items():
+                try:
+                    pod_metrics = parse_vdbench_output(logs)
```

Comment:
If i get it right, this function receive full pod log as argument. it might cause performance issues in case of very large logs - my suggestion is to limit number of lines, something like 100-200 last lines